### ISSUE TITLE
Name or random string used as ID within OID with Netscalers

### ISSUE THREAD

**Issue by RichiH** [prometheus/snmp_exporter#26]
As per IRC:

These blocks encode the length of what is then used as ID.

If the server name is 31 chars in length or less, decimal ASCII values are used in the OID. At length 32 or longer, the length of the ID in the block stays at 31 and a random string is used as ID.

An, outdated, MIB can be found at http://www.circitor.fr/Mibs/Files/NS-ROOT-MIB.mib

Example of the random id: `SNMPv2-SMI::enterprises.5951.4.1.3.1.1.59.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101`

**Comment by brian-brazil** [comment]
This is an OCTET STRING index.

**Comment by RichiH** [comment]
Actual return values for one server:

````
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.1.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = STRING: "INysq2p3ajfuzxacrtl1urazo24eabe"
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.2.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = IpAddress: <lolno>
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.3.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = INTEGER: 443
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.4.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = INTEGER: 14
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.5.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = INTEGER: 7
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.7.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = Gauge32: 1
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.8.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = Gauge32: 1
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.10.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = Gauge32: 0
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.30.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = Counter64: 6760369
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.31.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = Counter64: 162792393790
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.32.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = Counter64: 6760344
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.33.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = Counter64: 3157941205147
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.34.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = Counter64: 1273987380
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.35.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = Counter64: 2353133205
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.36.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = Counter64: 2427013
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.37.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = Gauge32: 0
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.38.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = Gauge32: 0
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.39.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = Gauge32: 0
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.40.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = Gauge32: 0
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.41.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = Gauge32: 2
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.42.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = Counter64: 0
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.43.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = STRING: "0"
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.44.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = STRING: "126"
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.45.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = STRING: "925"
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.46.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = STRING: "0"
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.47.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = Hex-STRING: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.48.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = Counter64: 6760369
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.54.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = Counter32: 0
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.56.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = Counter64: 2425369
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.58.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = STRING: "0"
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.59.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = STRING: <actual human readable name>
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.60.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = Gauge32: 0
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.61.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = Gauge32: 2
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.62.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = INTEGER: 100
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.63.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = Timeticks: (1545281200) 178 days, 20:26:52.00
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.64.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = INTEGER: 1
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.65.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = Counter64: 0
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.66.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = INTEGER: 0
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.67.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = Counter64: 4
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.68.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = Counter64: 0
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.69.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = INTEGER: 0
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.70.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = Gauge32: 0
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.71.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = Gauge32: 1
SNMPv2-SMI::enterprises.5951.4.1.3.1.1.72.31.73.78.121.115.113.50.112.51.97.106.102.117.122.120.97.99.114.116.108.49.117.114.97.122.111.50.52.101.97.98.101 = Gauge32: 0
````

**Comment by RichiH** [comment]
In my limited testing, the random values for server names longer than 31 chars have values between 45-122.

**Comment by brian-brazil** [comment]
Fixed by #28
